packages/template-retail-react-app/app/components/display-price/index.jsx

```diff
@@ -7,61 +7,193 @@
-                    })}
-                </Text>
-            )}
-            <Text
-                as={typeof discountPrice === 'number' ? 's' : 'b'}
-                ml={typeof discountPrice === 'number' ? 2 : 0}
-                fontWeight={discountPrice ? 'normal' : 'bold'}
-                {...basePriceProps}
-            >
-                {intl.formatNumber(basePrice, {
-                    style: 'currency',
-                    currency: currency || activeCurrency
-                })}
-            </Text>
-        </Skeleton>
+    const currentPriceText = intl.formatNumber(price, {
+        style: 'currency',
+        currency
+    })
+    return isRange ? (
+        <Text
+            as={as}
+            {...extraProps}
+            aria-label={intl.formatMessage(msg.ariaLabelCurrentPriceWithRange, {
+                currentPrice: currentPriceText
+            })}
+        >
+            {intl.formatMessage(msg.currentPriceWithRange, {
+                currentPrice: currentPriceText
+            })}
+        </Text>
+    ) : (
+        <Text
+            as={as}
+            aria-label={intl.formatMessage(msg.ariaLabelCurrentPrice, {
+                currentPrice: currentPriceText
+            })}
+        >
+            {intl.formatMessage(msg.currentPrice, {
+                currentPrice: currentPriceText
+            })}
+        </Text>
     )
 }
+CurrentPrice.propTypes = {
+    price: PropTypes.number.isRequired,
+    currency: PropTypes.string.isRequired,
+    as: PropTypes.string,
+    isRange: PropTypes.bool,
+    extraProps: PropTypes.object
+}
 
+/**
+ * Component that displays list price of a product with a11y
+ * @param price - price of the product
+ * @param as - an HTML tag or component to be rendered as
+ * @param isRange - show price as range or not
+ * @param props - extra props to be passed into Text Component
+ * @param extraProps - extra props to be passed into Text Component
+ * @returns {JSX.Element}
+ */
+const ListPrice = ({price, isRange = false, as = 's', currency, ...extraProps}) => {
+    const intl = useIntl()
+    const listPriceText = intl.formatNumber(price, {
+        style: 'currency',
+        currency
+    })
+
+    return isRange ? (
+        <Text
+            as={as}
+            {...extraProps}
+            aria-label={intl.formatMessage(msg.ariaLabelListPriceWithRange, {
+                listPrice: listPriceText || ''
+            })}
+            color="gray.600"
+        >
+            {intl.formatMessage(msg.listPrice, {
+                listPrice: listPriceText
+            })}
+        </Text>
+    ) : (
+        <Text
+            as={as}
+            aria-label={intl.formatMessage(msg.ariaLabelListPrice, {
+                listPrice: listPriceText || ''
+            })}
+            color="gray.600"
+        >
+            {intl.formatMessage(msg.listPrice, {
+                listPrice: listPriceText
+            })}
+        </Text>
+    )
```

Comment:
Similar to my previous comment, we can write this as:

```suggestion
    const ariaLabel = intl.formatMessage(
        isRange ? msg.ariaLabelListPriceWithRange : msg.ariaLabelListPrice,
        {
            listPrice: listPriceText || ''
        }
    )
    const message = intl.formatMessage(msg.listPrice, {
        listPrice: listPriceText
    })

    return (
        <Text as={as} {...extraProps} aria-label={ariaLabel} color="gray.600">
            {message}
        </Text>
    )
```